app/components/github-connect.js

```diff
@@ -0,0 +1,37 @@
+import Ember from 'ember';
+
+const {
+  Component,
+  computed,
+  get
+} = Ember;
+
+/**
+  The github-connect component is used to to connect to github
+
+  ## default usage
+
+  ```handlebars
+    {{github-connect clientId=clientId scope=scope state=state redirectUri=redirectUri}}
+  ```
+
+  @module Component
+  @class github-connect
+  @extends Ember.Component
+  @public
+ */
+
+export default Component.extend({
+  tagName: 'a',
+  classNames: ['button', 'default'],
+  attributeBindings: ['url:href'],
+
+  clientId: null,
+  scope: null,
+  state: null,
+  redirectUri: null,
+
+  url: computed('clientId', 'scope', 'state', 'redirectUrl', function() {
```

Comment:
@terrancebryant yeah so it would be something like 

```
init() {
  this._super(...arguments);
  let { clientId, scope, state, redirectUrl } = this.getProperties('clientId', 'scope', 'state', 'redirectUrl');
  set(this, 'url', `${url}?scope=${scope}&client_id=${clientId}....`)
}
```